oak_restricted_kernel/src/syscall/switch_process.rs

```diff
@@ -39,27 +34,13 @@ pub fn syscall_unstable_switch_proccess(buf: *mut c_void, count: c_size_t) -> !
     let application = crate::payload::Application::new(copied_elf_binary.into_boxed_slice())
         .expect("failed to parse application");
 
-    let (base_pml4, encrypted) = crate::BASE_L4_PAGE_TABLE
-        .get()
-        .expect("base l4 table should be set");
-    // Ensure the new page table is not dropped.
-    let pml4 = Box::leak(Box::new(base_pml4.clone()));
-
-    let pml4_frame = {
-        let phys_addr = {
-            let addr = &*pml4 as *const PageTable;
-            let pt_guard = crate::PAGE_TABLES.lock();
-            let pt = pt_guard.get().expect("failed to get page tables");
-            pt.translate_virtual(VirtAddr::from_ptr(addr))
-                .expect("failed to translate virtual page table address")
-        };
-        PhysFrame::from_start_address(phys_addr)
-            .expect("couldn't get the physical frame for page table address")
-    };
+    // Ensure the new process is not dropped. Safety: the ELF file must be valid.
+    let process = Box::leak(Box::new(
+        // Safety: application must be valid.
```

Comment:
nit: the safety statement here should not specify the requirement, but why we believe the requirements is met and it is therefore OK to use `unsafe`. Also, elswhere in the PR where unsafe functions are called.

Reply:
updated, thanks.